ui/pages/index.js

```diff
@@ -53,7 +54,9 @@ class Index extends PureComponent {
                     <CurrencyRateProvider>
                       <MetamaskNotificationsProvider>
                         <TokenRatesProvider>
-                          <Routes />
+                          <TokenDetectionProvider>
```

Comment:
inject `detectedTokens` as provider , this will add to our global state the object below:
```
{ '0x1': { detected token ...} , ....}
```